Admin/AbstractAdmin.php

```diff
@@ -1490,13 +1509,28 @@ public function getLabel()
 
     /**
      * @param bool $persist
+     *
+     * @deprecated since 3.x, to be removed in 4.0.
      */
     public function setPersistFilters($persist)
     {
+        @trigger_error(
+            'The '.__METHOD__.' method is deprecated since version 3.x and will be removed in 4.0.',
+            E_USER_DEPRECATED
+        );
+
         $this->persistFilters = $persist;
```

Comment:
Still not BC because it worked before when the session wasn't set on the admin.

Anyway, I think this should work with the old content...
Just add a check for `$this->persistFilters !== false` [here](https://github.com/yann-eugone/SonataAdminBundle/blob/c7d317e9f31180ce523409d8692937d3512197f0/Admin/AbstractAdmin.php#L788) and don't forget to add a `NEXT_MAJOR` comment too.